packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -424,11 +577,65 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties.add(DiagnosticsProperty<Object>('owner', owner, level: parent != null ? DiagnosticLevel.hidden : DiagnosticLevel.info, defaultValue: null));
-    properties.add(DiagnosticsProperty<dynamic>('creator', debugCreator, defaultValue: null, level: DiagnosticLevel.debug));
-    properties.add(DiagnosticsProperty<String>('engine layer', describeIdentity(_engineLayer)));
+    properties.add(DiagnosticsProperty<Object?>('creator', debugCreator, defaultValue: null, level: DiagnosticLevel.debug));
+    if (_engineLayer != null) {
+      properties.add(DiagnosticsProperty<String>('engine layer', describeIdentity(_engineLayer)));
+    }
+    properties.add(DiagnosticsProperty<int>('handles', debugHandleCount));
   }
 }
 
+/// A handle to prevent a [Layer]'s platform graphics resources from being
+/// disposed.
+///
+/// [RenderObject]s that need to hold more than one [Layer] may use the
+/// [LayerHolder] class to help manage the lifetime of additional layers.
+///
+/// [Layer] objects retain native resourses such as [EngineLayer]s and [Picture]
+/// objects. These objects may in turn retain large chunks of texture memory,
+/// either directly or indirectly.
+///
+/// The layer's native resources must be retained as long as there is some
+/// object that can add it to a scene. Typically, this is either its
+/// [Layer.parent] or an undisposed [RenderObject] that will append it to a
+/// [ContainerLayer]. Layers automatically hold a handle to their children, and
+/// RenderObjects automatically hold a handle to their [RenderObject.layer] as
+/// well as any [PictureLayer]s that they paint into using the
+/// [PaintingContext.canvas]. A layer automatically releases its resources once
+/// at least one handle has been acquired and all handles have been disposed.
+/// [RenderObject]s that create additional layer objects must manually manage
+/// the handles for that layer similarly to the implementation of
+/// [RenderObject.layer].
+///
+/// Layers can be reused even after their native resources has been released,
+/// since they are mutable objects and the native resources can be reset on
+/// subsequent frames.
+class LayerHandle {
```

Comment:
reviewers: I'm somewhat tempted to make this a `mixin` on `RenderObject` so it can override and handle `dispose` for you.  The problem is that it would mean a RenderObject could only easily handle one additional layer, and we'd have to give the layer a weird property name like `additionalLayer` or something.

The framework only needs one additional layer for any of the RenderObjects that tests have picked up, but I'm a bit put off by the naming weirdness and the odd restriction. I'm open to ideas though!